### ISSUE TITLE
1835 - PR loses director

### ISSUE THREAD

**Issue by madoar** [Rails-18xx/Rails#320]
In certain situations where the PR director sells one or more share the director "disappears" from the game.
By disappears I mean that no player receives the director share and the game breaks.
The effect has been discovered in the current version (2.3.0) of Rails.

## Error Reproduction
1)  load: [1835_20201220_2109_Marc.rails.zip](https://github.com/Rails-18xx/Rails/files/5724383/1835_20201220_2109_Marc.rails.zip)
2)  sell 40% PR shares as player "Marc"
3)  now none of the players has the PR director

**Comment by erik-vos** [comment]
Yes, I have also seen that happen very recently.

I'm currently working on implementing proper Bankruptcy handling in games that don't terminate if a bankruptcy occurs.
Thereby it often turns out that emergency train buying isn't bugfree as well. Done are 18Scan, SOH (in 2.3.0) and 18EU (in a later commit), and I can't exclude that my work on these games has had bad effects on 1835; in my first look at that game, after constructing an emergency case, it even crashed. 

The trouble for me is, that I find the v2.0 code in ShareSellingRound and its called methods, in particular PlayerShareUtils, difficult to follow, so I'm struggling to understand what exactly is going on there. For 2.3.0 I had already changed some code in and around ShareSellingRound, and perhaps that has caused this problem. I got it fixed, but now some games in the test set fail, so I'm not done yet. I'm actually on the verge of rewriting _and documenting_ much of that code, so at least I can understand it. 

I'm continuously adding emergency and bankruptcy cases to the test set, so hopefully adverse code changes in this area will in the future be detected in an early stage.
